src/strategies/MangroveOrder.sol

```diff
@@ -60,8 +47,39 @@ contract MangroveOrder is Forwarder, IOrderLogic {
   }
 
   ///@inheritdoc IOrderLogic
-  function setAdditionalGasreq(uint additionalGasreq_) external onlyAdmin {
-    additionalGasreq = additionalGasreq_;
+  ///@dev We also allow Mangrove to call this so that it can part of an offer logic.
+  function setExpiry(IERC20 outbound_tkn, IERC20 inbound_tkn, uint offerId, uint date)
+    public
+    mgvOrOwner(outbound_tkn, inbound_tkn, offerId)
+  {
+    expiring[outbound_tkn][inbound_tkn][offerId] = date;
+  }
+
+  ///@notice updates an offer on Mangrove
+  ///@dev this can be used to update price of the resting order
+  ///@param outbound_tkn outbound token of the offer list
+  ///@param inbound_tkn inbound token of the offer list
+  ///@param wants new amount of `inbound_tkn` offer owner wants
+  ///@param gives new amount of `outbound_tkn` offer owner gives
+  ///@param pivotId pivot for the new rank of the offer
+  ///@param offerId the id of the offer to be updated
+  function updateOffer(IERC20 outbound_tkn, IERC20 inbound_tkn, uint wants, uint gives, uint pivotId, uint offerId)
+    external
+    payable
+    onlyOwner(outbound_tkn, inbound_tkn, offerId)
+  {
+    OfferArgs memory args;
+
+    // funds to compute new gasprice is msg.value. Will use old gasprice if no funds are given
+    args.fund = msg.value; // if inside a hook (Mangrove is `msg.sender`) this will be 0
+    args.outbound_tkn = outbound_tkn;
+    args.inbound_tkn = inbound_tkn;
+    args.wants = wants;
+    args.gives = gives;
+    args.gasreq = offerGasreq();
+    args.pivotId = pivotId;
+    args.noRevert = false; // will throw if Mangrove reverts
+    _updateOffer(args, offerId);
```

Comment:
Internal `_updateOffer` is no longer exposed by Forwarder strats. We need to expose it here. This is the opportunity to hide `gasprice` and `gasreq` which are not user controlled parameters.